File: connecttoapilibrary/src/main/java/com/sec/connecttoapilibrary/onlinefm/NewFMListData.java
package com.sec.connecttoapilibrary.onlinefm;

import android.content.Context;

import com.sec.connecttoapilibrary.onlinefm.liveRadioData.Station;
import com.sec.connecttoapilibrary.onlinefm.liveRadioData.StationProgram;
import com.sec.connecttoapilibrary.onlinefm.defultonlineinfolistener.ObserverDefaultLiveRadioListener;
import com.sec.connecttoapilibrary.onlinefm.defultonlineinfolistener.ObserverDefaultLiveRadioDefaultLiveRadioListenerManager;
import com.sec.connecttoapilibrary.onlinefm.updataLiveRadioUIListener.ObserverLiveRadioLiveRadioUIListenerManager;

import java.util.List;
import java.util.Map;

/**
 * Created by SRC-TJ-MM-BinYang on 2018/1/31.
 */

public class NewFMListData implements ObserverDefaultLiveRadioListener {

    private Context mContext = null;
    private List<Station> mStations = null;
    private Map<Integer, List<StationProgram>> map = null;
    private List<Station> mCenterStations;
    private Map<Integer, List<StationProgram>> centermap;
    private FMListData mFMlistData;

    public NewFMListData(Context context){
        mContext = context;
        ObserverDefaultLiveRadioDefaultLiveRadioListenerManager.getInstance().add(this);
        mFMlistData = new FMListData(mContext);
    }

    @Override
    public void observerUpDataLocalList(List<Station> mStations, Map<Integer, List<StationProgram>> map) {
        this.mStations = mStations;
        this.map = map;
        doUpdateUILocalList();
    }

    @Override
    public void observerUpDataCenterList(List<Station> mCenterStations, Map<Integer, List<StationProgram>> center_map) {
        this.mCenterStations = mCenterStations;
        this.centermap = center_map;
        doUpdateUICenterList();
    }

    @Override
    public void observerUpDataDifferentInfo(List<Station> mStations, Map<Integer, List<StationProgram>> map) {
        mFMlistData.addNetInfo(mStations,map);
        ObserverLiveRadioLiveRadioUIListenerManager.getInstance().notifyDifferentInfoObserver(mStations, map);
    }

    public void doUpdateUILocalList() {
        mFMlistData.makeLocationList(FMListData.TYPE_LOCAL_LIST, mStations, map);
    }
    public void doUpdateUICenterList() {
        mFMlistData.makeLocationList(FMListData.TYPE_CENTER_LIST, mStations, map);
    }
}

File: myonlinefm/src/main/java/com/sec/myonlinefm/NewFMListData.kt
package com.sec.myonlinefm

import android.content.Context

import com.sec.myonlinefm.data.Station
import com.sec.myonlinefm.data.StationProgram
import com.sec.myonlinefm.onlineinfolistener.ObserverListener
import com.sec.myonlinefm.onlineinfolistener.ObserverListenerManager
import com.sec.myonlinefm.updataUIListener.ObserverUIListenerManager

/**
 * Created by SRC-TJ-MM-BinYang on 2018/1/31.
 */

class NewFMListData (context: Context) : ObserverListener {

    var mContext : Context? = null
    var mStations : MutableList<Station>? = null
    private var mMap : MutableMap<Int, List<StationProgram>> ? = null
    private var mCenterStations : MutableList<Station>? = null
    private var mCenterMap : MutableMap<Int, List<StationProgram>> ? = null
    var mPlayer : OnLineFMConnectManager
    private var mFMListData : FMListData

    init {
        mContext = context
        mPlayer = OnLineFMConnectManager.mMainInfoCode!!
        ObserverListenerManager.getInstance().add(this)
        mFMListData = FMListData(mContext)
    }
    override fun observerLiveRadioLocalUpData(stations : MutableList<Station>, map : MutableMap<Int, List<StationProgram>>) {
        this.mStations = stations
        this.mMap = map
        mFMListData.makeLocationList(mStations,mMap, FMListData.TYPE_LOCAL)
    }

    override fun observerLiveRadioCenterUpData(mStations : MutableList<Station>, map : MutableMap<Int, List<StationProgram>>) {
        this.mCenterStations = mStations
        this.mCenterMap = map
        mFMListData.makeLocationList(mCenterStations,mCenterMap, FMListData.TYPE_CENTER)
    }

    override fun observerUpDataDifferentInfo(mStations : MutableList<Station>, map : MutableMap<Int, List<StationProgram>>) {
        mFMListData.addNetInfo(mStations,map)
        ObserverUIListenerManager.getInstance().notifyDifferentInfoObserver(mStations, map)
    }

//    fun doUpdateUI() {
//        mFMListData.makeLocationList(mStations, map,mCenterStations,centerMap)
//        mFMListData.makeLocationList(mStations, map,mCenterStations,centerMap);
//    }
    fun doConnectApi(){
        mPlayer.getOnLineFM()
    }
    fun stopConnectApi(){
//        mPlayer.StopOnLineFM();
    }
}
